pkg/client/monitoring/v1alpha1/types.go

```diff
@@ -80,6 +80,8 @@ type PrometheusSpec struct {
 	RoutePrefix string `json:"routePrefix,omitempty"`
 	// Storage spec to specify how storage shall be used.
 	Storage *StorageSpec `json:"storage,omitempty"`
+	// Pvc A pvc spec to be used by the Prometheus statefulsets.
+	Pvc *v1.PersistentVolumeClaim `json:"pvc,omitempty"`
```

Comment:
Maybe rather `volumeClaimTemplate`? And it would probably make sense to move this field into the storage spec, and then just deprecate all other fields in the `StorageSpec`.